django/db/migrations/executor.py

```diff
@@ -77,6 +76,24 @@ def _create_project_state(self, with_applied_migrations=False):
             for migration, _ in full_plan:
                 if migration in applied_migrations:
                     migration.mutate_state(state, preserve=False)
+                elif self.loader.is_initial_migration(migration):
+                    fake_initial = False
+                    queue = deque([migration])
+                    while queue and not fake_initial:
+                        m = queue.popleft()
+                        for child in self.loader.graph.node_map[m.app_label, m.name].children:
+                            child_migration = self.loader.graph.nodes[child]
+                            if child_migration in applied_migrations:
+                                fake_initial = True
+                                break
+                            if self.loader.is_initial_migration(child_migration):
```

Comment:
I wasn't 100% sure what was the purpose of this. (I didn't quite follow the discussion from the ticket of the cross-app dependency.) 

Two points: 

* Again a comment here would help. `# Re-apply across apps...` (or something)
* There's no test covering this case: If I comment out these lines nothing fails.